src/effects/native/utilityeffect.cpp

```diff
@@ -0,0 +1,165 @@
+		//rampgain ramps down with rampstep till zero.
+		pState->ramp_state = UtilityGroupState::RAMPDOWN;
+		//set previous mute state to mute
+		pState->previous_mute_state = true;
+	}
+
+	// mute->unmute state transition
+	if ((!pState->previous_mute_state)  && (mute_state) ) {
+		//rampgain ramps up from zero to 1.0
+		pState->ramp_state = UtilityGroupState::RAMPUP;
+		//set previous mute state to unmute
+		pState->previous_mute_state = false;
+	}
+
+	// disable->enable state transition
+	if (pState->previous_enable_state == EffectProcessor::DISABLED && enableState == EffectProcessor::ENABLED) {
+		//ramp gain ramps up from zero to 1.0
+		pState->ramp_state = UtilityGroupState::RAMPUP;
+		//set previous enable state to enable
+		pState->previous_enable_state = EffectProcessor::ENABLED;
+	}
+
+	// enable->disable state transition
+	if ((pState->previous_enable_state == EffectProcessor::ENABLED) && (enableState == EffectProcessor::DISABLED)) {
+		//ramp gain ramps down with rampstep till zero.
+		pState->ramp_state = UtilityGroupState::RAMPDOWN;
+		//set previous enable state to disable
+		pState->previous_enable_state = EffectProcessor::DISABLED;
+	}
+
+	// increment or decrement rampgain depending on rampstate.
+	if(pState->ramp_state == UtilityGroupState::RAMPDOWN) {
+		pState->ramp_gain-=ramp_step;
+	}
+
+	if(pState->ramp_state == UtilityGroupState::RAMPUP) {
+		pState->ramp_gain+=ramp_step;
+	}
+
+	//add ramp gain guard to stop ramping.
+	if (pState->ramp_gain > 1.0) {
+		pState->ramp_gain =1.0;
+		pState->ramp_state = UtilityGroupState::RAMPNO;
+	}
+
+	if (pState->ramp_gain < 0.0) {
+		pState->ramp_gain =0.0;
+		pState->ramp_state = UtilityGroupState::RAMPNO;
+	}
+
+}
+
+void UtilityEffect::processChannel(const ChannelHandle& handle,
+                                      UtilityGroupState* pState,
+                                      const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                      const unsigned int numSamples,
+                                      const unsigned int sampleRate,
+                                      const EffectProcessor::EnableState enableState,
+                                      const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(handle);
+    Q_UNUSED(groupFeatures);
+
+
+    CSAMPLE effect_gain = m_pKnobA ?
+    		m_pKnobA->value() : 0.0;
+
+    bool current_mute_state = m_pSwitchA->toBool();
+
+    //if effect is disabled and ramping complete return,
+    if ((enableState == EffectProcessor::DISABLED)&&(pState->ramp_state == UtilityGroupState::RAMPNO)) {
+        return;
+    }
+
+	//increase and decrease ramp factor from 0.0 to 1.0 and back with this step.
+	CSAMPLE ramp_step = (pState->ramp_time_in_seconds * numSamples)/(2.0*sampleRate);
+
+	UtilityEffect::rampGainCalculation(pState, current_mute_state, ramp_step, enableState );
+
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
```

Comment:
we have vectorized functions for this case. SampleUtil::copyWithRampingGain() or similar. Pleas try to make use of this.